game/transfers.py

```diff
@@ -571,6 +571,13 @@ def __init__(self, game: Game, player: bool) -> None:
     def __iter__(self) -> Iterator[TransferOrder]:
         yield from self.pending_transfers
 
+    def __update_eventstream__(self) -> None:
```

Comment:
Why is this a dunder method? If your intent was just to signal "private" (oh how I wish Python would admit that private methods are useful), that's a single underscore prefix for the name. Dunder methods are reserved by python for language defined behaviors (and there actually are behavioral differences between all three name "styles").

Moreover, I don't think this object should be the one sending the events. The methods below should accept a `GameUpdateEvents`, since the caller typically already has an event object that it's building up to send.

A thing I'd forgotten to mention in the other review (because I forgot about it), a better way to do these quick, ephemeral event messages is:

```python
with EventStream.event_context() as events:
    events.update_supply_routes()
```

Not what you'll want _here_, since anything happening during turn processing should go to the turn's event object, but that pattern is useful in the UI.

Reply:
What you're suggesting is not possible, creates circular dependency because originally that's how I wanted to tackle the problem, good thing you pointed out the dunder method, don't know what I was thinking, should be a single underscore indeed...

Reply:
Wdym circular dependency?

Reply:
If you want to add events to method signatures, you're gonna need to import `GameUpdateEvents` on global level, that's gonna cause a crash making python complain about not being able to import a partially initialized module.

Reply:
Do the import inside `if TYPE_CHECKING`. Should be able to find that pattern in just about every file we have if you want an example. Python's a pain in the ass about it but it is what it is.

Reply:
Ok in that case I'll convert back to draft and give it another shot tomorrow, though I'm afraid the solution you suggested is gonna be more complicated since I'll have to assure all calls being made to the applicable methods pass along a `GameUpdateEvents` object. On top of that the calling objects may not always have a `GameUpdateEvents` object at hand to pass along.

Either way, I'll have another look at it and let you if I find a better spot for pushing the updates.